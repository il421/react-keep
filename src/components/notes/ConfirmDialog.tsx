import React, { Dispatch } from "react";
import { connect } from "react-redux";
import { handleRemoveNote } from "../../actions/notes";

/**
 * This component works only as a part of Note for now.
 */

interface ConfirmDialogProps extends Pick<HTMLElement, "className">{
  id: string;
  removeNote: (id: string) => void;
  closeDialog: () => void;
}

const ConfirmDialog: React.FunctionComponent<ConfirmDialogProps> = ({ className, closeDialog, removeNote, id }) => {
  return (
    <div className={className}>
      <button type="button"
        onClick={() => {
          removeNote(id);
          closeDialog();
        }
        }
      >Yes</button>
      <button type="button" onClick={closeDialog}>No</button>
    </div>
  );
};


const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  removeNote: (id: string) => dispatch(handleRemoveNote(id))
});


export default connect(undefined, mapDispatchToProps)(ConfirmDialog);
