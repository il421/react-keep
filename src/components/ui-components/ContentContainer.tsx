import React, {CSSProperties} from "react";
import "../../styles/components/common/_content-conteiner.scss";

interface ContentContainerProps extends Partial<Pick<HTMLElement, "className">> {
  style?: CSSProperties
}

export class ContentContainer extends React.PureComponent<ContentContainerProps> {
  render() {
    return (
      <div
        className={`content-container ${ this.props.className }`}
        style={this.props.style}
      >{ this.props.children }
      </div>
    );
  }
}
