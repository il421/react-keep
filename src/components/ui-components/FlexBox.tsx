import React, { CSSProperties } from "react";
import { AlignItems, FlexWrap, JustifyContent } from "../../common/variables";

interface FlexBoxProps extends Partial<Pick<HTMLElement, "className">> {
  vertical?: boolean;
  justifyContent: JustifyContent;
  alignItems?: AlignItems;
  flexWrap?: FlexWrap;
  width?: string;
}

/**
 * Flex box components is latterly is flex box styled block.
 * @param FlexBoxProps
 */

export class FlexBox extends React.PureComponent<FlexBoxProps> {
  private style: CSSProperties = {
    display: "flex",
    width: this.props.width,
    flexDirection: this.props.vertical ? "column" : "row",
    justifyContent: this.props.justifyContent,
    alignItems: this.props.alignItems ?? AlignItems.inherit,
    flexWrap: this.props.flexWrap ?? FlexWrap.inherit,
  };

  render() {
    return (
      <div style={this.style} className={this.props.className}>
        {this.props.children}
      </div>
    );
  }
}
