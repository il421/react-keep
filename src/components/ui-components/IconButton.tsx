import React from "react";
import {
  FontAwesomeIcon,
  FontAwesomeIconProps
} from "@fortawesome/react-fontawesome";
import "../../styles/ui-components/_icon-button.scss";

interface IconButtonProps extends FontAwesomeIconProps {
  onClick: (evt?: React.MouseEvent) => void;
  className?: string;
  disabled?: boolean;
}

export class IconButton extends React.PureComponent<IconButtonProps> {
  render() {
    const { onClick, className, disabled, ...rest } = this.props;
    return (
      <button
        className={`${className} icon-button`}
        onClick={onClick}
        disabled={disabled}
      >
        <FontAwesomeIcon {...rest} />
      </button>
    );
  }
}
