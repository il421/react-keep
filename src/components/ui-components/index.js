export * from "./ContentContainer";
export * from "./FlexBox";
export * from "./IconButton";
export * from "./LinkButton";
export * from "./ConfirmButton";
export * from "./UserPhoto";
