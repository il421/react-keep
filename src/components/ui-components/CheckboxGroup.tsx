import React from "react";

import { JustifyContent } from "../../common";
import "../../styles/ui-components/_login-button.scss";
import { FlexBox } from "./FlexBox";

export interface Option {
  id: string;
  value: string;
}
export interface CheckboxGroupProps {
  options: Option[];
  selectedKeys: string[];
  onChange: (selectedKeys: string[]) => void;
  classNames?: {
    option?: string;
    checkbox?: string;
    label?: string;
  };
}

export class CheckboxGroup extends React.PureComponent<CheckboxGroupProps> {
  private handleOnChange = (selectedKey: string) => {
    let selectedKeys: string[] = this.props.selectedKeys || [];

    const isAlreadySelected = selectedKeys.includes(selectedKey);
    if (isAlreadySelected) {
      selectedKeys = selectedKeys.filter(x => x !== selectedKey);
    } else {
      selectedKeys = [...selectedKeys, selectedKey];
    }
    if (this.props.onChange) {
      this.props.onChange(selectedKeys);
    }
  };

  render() {
    const { options, classNames, selectedKeys } = this.props;

    return (
      <>
        {options.map((o: Option) => {
          const checked = selectedKeys.includes(o.id);

          return (
            <FlexBox
              key={o.id}
              justifyContent={JustifyContent.start}
              className={classNames?.option}
            >
              <div className={classNames?.checkbox}>
                <input
                  ref={o.id}
                  id={o.id}
                  value={o.id}
                  type="checkbox"
                  defaultChecked={checked}
                />
                <label
                  htmlFor={o.id}
                  onClick={() => this.handleOnChange(o.id)}
                />
              </div>

              <span className={classNames?.label}>{o.value}</span>
            </FlexBox>
          );
        })}
      </>
    );
  }
}
