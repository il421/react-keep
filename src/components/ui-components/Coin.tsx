import React, { memo } from "react";
import ReactTooltip from "react-tooltip";
import "../../styles/ui-components/_coin.scss";
import { getRandomColor, getShortText, JustifyContent } from "../../common";
import { FlexBox } from "./FlexBox";

export interface CoinBaseProps {
  email: string | null;
  name: string | null;
  url?: string | null;
  showTooltip?: boolean;
  className?: string;
}
const CoinBase: React.FunctionComponent<CoinBaseProps> = ({
  email,
  name,
  url,
  showTooltip,
  className,
}) => {
  return (
    <div data-tip={name ?? email}>
      <FlexBox
        justifyContent={JustifyContent.center}
        className={`coin ${className}`}
      >
        {url ? (
          <img
            src={url}
            alt={`Photo of ${name ?? email}`}
            width={25}
            height={25}
          />
        ) : (
          <span style={{ backgroundColor: getRandomColor() }}>
            {getShortText(name ?? email!, 2, false)}
          </span>
        )}
      </FlexBox>
      {showTooltip && (
        <ReactTooltip
          place="top"
          type="dark"
          effect="solid"
          className="coin__tooltip"
        />
      )}
    </div>
  );
};

export const Coin = memo(CoinBase);
