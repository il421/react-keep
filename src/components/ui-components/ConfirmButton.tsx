import React from "react";
import { BeatLoader } from "react-spinners";
import { Colors, JustifyContent } from "../../common";
import { FlexBox } from "./FlexBox";
import "../../styles/ui-components/_login-button.scss";

export interface ConfirmButtonProps
  extends Partial<Pick<HTMLButtonElement, "disabled" | "className" | "id">> {
  text: string;
  loading?: boolean;
  type?: "submit" | "button";
  onCLick?: (evt?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  wrapperClassName?: string;
}

export class ConfirmButton extends React.PureComponent<ConfirmButtonProps> {
  render() {
    const {
      type = "submit",
      loading,
      className,
      text,
      disabled = false,
      onCLick,
      id,
      wrapperClassName,
    } = this.props;
    return (
      <FlexBox
        className={wrapperClassName}
        justifyContent={JustifyContent.center}
      >
        {!loading ? (
          <button
            id={id}
            className={className}
            type={type}
            disabled={disabled}
            onClick={onCLick}
          >
            {text}
          </button>
        ) : (
          <BeatLoader
            size={10}
            color={Colors.fresh}
            css={"height: 42px; display: flex; align-items: center"}
          />
        )}
      </FlexBox>
    );
  }
}
