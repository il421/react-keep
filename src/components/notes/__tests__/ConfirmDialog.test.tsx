import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { ConfirmDialogProps, ConfirmDialog } from "../ConfirmDialog";
import { ConfirmButtonProps } from "../../ui-components";

let wrapper: ReactWrapper<ConfirmDialogProps>, props: ConfirmDialogProps;
const btnNames = {
  confirmButtonText: "confirmTest",
  cancelButtonText: "cancelTest",
};
beforeEach(() => {
  props = {
    className: "confirm-dialog",
    closeDialog: jest.fn(),
    handleConfirm: jest.fn(),
    buttonsProps: {
      confirmButtonText: btnNames.confirmButtonText,
      cancelButtonText: btnNames.cancelButtonText,
    },
  };
  wrapper = mount(<ConfirmDialog {...props} />);
});

test("should render ConfirmDialog correctly", () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

test("should call correct method by clicking actions buttons (with new btn names)", () => {
  const deleteBtn = wrapper.find("#test-confirm-dialog-delete-btn").hostNodes();
  const dontBtn = wrapper.find("#test-confirm-dialog-dont-btn").hostNodes();

  expect(deleteBtn.exists()).toBeTruthy();
  expect(deleteBtn.text()).toBe(btnNames.confirmButtonText);
  deleteBtn.simulate("click");
  expect(props.handleConfirm).toBeCalled();

  expect(dontBtn.exists()).toBeTruthy();
  expect(dontBtn.text()).toBe(btnNames.cancelButtonText);
  dontBtn.simulate("click");
  expect(props.closeDialog).toBeCalled();
});
