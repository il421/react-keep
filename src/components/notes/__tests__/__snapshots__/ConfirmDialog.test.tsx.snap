// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmDialog correctly 1`] = `
"<ConfirmDialog className=\\"confirm-dialog\\" closeDialog={[Function: mockConstructor]} handleConfirm={[Function: mockConstructor]} buttonsProps={{...}}>
  <FlexBox className=\\"confirm-dialog confirm-dialog\\" justifyContent=\\"center\\" alignItems=\\"center\\">
    <div style={{...}} className=\\"confirm-dialog confirm-dialog\\">
      <FlexBox className=\\"confirm-dialog__wrapper\\" justifyContent=\\"space-around\\">
        <div style={{...}} className=\\"confirm-dialog__wrapper\\">
          <ConfirmButton id=\\"test-confirm-dialog-delete-btn\\" type=\\"button\\" text=\\"confirmTest\\" wrapperClassName=\\"confirm-dialog__button\\" onCLick={[Function: onCLick]}>
            <FlexBox className=\\"confirm-dialog__button\\" justifyContent=\\"center\\">
              <div style={{...}} className=\\"confirm-dialog__button\\">
                <button id=\\"test-confirm-dialog-delete-btn\\" className=\\"login-button\\" type=\\"button\\" disabled={false} onClick={[Function: onCLick]}>
                  confirmTest
                </button>
              </div>
            </FlexBox>
          </ConfirmButton>
          <ConfirmButton id=\\"test-confirm-dialog-dont-btn\\" type=\\"button\\" text=\\"cancelTest\\" wrapperClassName=\\"confirm-dialog__button\\" onCLick={[Function: onCLick]}>
            <FlexBox className=\\"confirm-dialog__button\\" justifyContent=\\"center\\">
              <div style={{...}} className=\\"confirm-dialog__button\\">
                <button id=\\"test-confirm-dialog-dont-btn\\" className=\\"login-button\\" type=\\"button\\" disabled={false} onClick={[Function: onCLick]}>
                  cancelTest
                </button>
              </div>
            </FlexBox>
          </ConfirmButton>
        </div>
      </FlexBox>
    </div>
  </FlexBox>
</ConfirmDialog>"
`;
